config/manager.go

```diff
@@ -235,6 +237,11 @@ func (f *FilesystemConfigManager) newKey() error {
 		return err
 	}
 
+	err = f.fsHandle.CreateFile(f.vault, internal.PermUserRW)
```

Comment:
Wouldn't make sense instead of having this to add `chmod` into the `WriteFile`, to fix all the places the permissions? if this was made to set the current permissions on the file.
Because the same problem is with `f.save(*c)`. If the file exists it will not change the permissions.